api/parse_request.go

```diff
@@ -0,0 +1,194 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/qfs"
+	"github.com/qri-io/qri/api/util"
+	"github.com/qri-io/qri/base/fill"
+	"github.com/qri-io/qri/dsref"
+	"github.com/qri-io/qri/lib"
+	"gopkg.in/yaml.v2"
+)
+
+// parseGetParamsFromRequest parse the form from the request to create `lib.GetParams`
+func parseGetParamsFromRequest(r *http.Request, p *lib.GetParams) error {
```

Comment:
co-locates all methods that pull params from request, except for the method defined on `params.List`

Definite argument can be made for getting rid of the `params.ListParams` interface, and moving the `ListParamsFromRequest` method here.